docs/user_guide/building_blocks/categories.md

```diff
@@ -1,3 +1,495 @@
+Work in Progess! This guide has not been finalized and the presented code has not yet been tested.
+
+TODO: make the code testable
+
 # Categories
 
-TODO
+In Columnflow, there are many tools to create a complex and flexible categorization of all analysed
+events. This guide presents, how to implement a set of categories in Columnflow and presents an
+showcases, how to use the resulting categories via the
+{py:class}`~columnflow.tasks.yields.CreateYieldsTable` task.
```

Comment:
I feel like it's much better to use the yields table as an example here, because you actually see the impact of your categorization on your final yields, which makes it easy to cross-check whether the categorization works as intended (e.g. sub-categories should sum up to the yield of an inclusive category). This could be missed quite easily when looking at plots instead.

Of course, this is not a replacement for documentation of the `CreateYieldsTable` and it should still (and hopefully is) mentioned in the guide that you can also use categories for e.g. plotting